clients/drcachesim/tracer/raw2trace.h

```diff
@@ -388,6 +391,22 @@ struct trace_metadata_reader_t {
     check_entry_thread_start(const offline_entry_t *entry);
 };
 
+#ifdef BUILD_PT_POST_PROCESSOR
+/**
+ * Functions for decoding and verifying raw memtrace data headers.
+ */
+struct ktrace_metadata_reader_t {
+    static bool
+    is_pt_metadata_header(const syscall_pt_entry_t *entry);
```

Comment:
I would expect these function operating on this syscall_pt_entry_t type to be part of the syscall_pt_entry_t struct, not some separate struct.